crypto/evp/evp_enc.c

```diff
@@ -962,13 +965,72 @@ int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
 int EVP_CIPHER_CTX_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg, void *ptr)
 {
     int ret;
+    size_t sz, n = 0;
+    OSSL_PARAM params[3];
 
-    if (!ctx->cipher) {
+    if (ctx == NULL || ctx->cipher == NULL) {
         EVPerr(EVP_F_EVP_CIPHER_CTX_CTRL, EVP_R_NO_CIPHER_SET);
         return 0;
     }
+    if (ctx->cipher->prov == NULL)
+        goto legacy;
+    if (ctx->cipher->ctx_get_params == NULL
+        || ctx->cipher->ctx_set_params == NULL)
+        return 0;
+    switch (type) {
+    case EVP_CTRL_GCM_SET_IV_FIXED:
+        params[n++] =
+            OSSL_PARAM_construct_octet_string(OSSL_CIPHER_AEAD_TLS1_IV_FIXED,
+                                              ptr, (size_t)arg);
+        params[n++] = OSSL_PARAM_construct_end();
+        return ctx->cipher->ctx_set_params(ctx->provctx, params);
```

Comment:
These two lines are common in all cases, why not have them after the switch?

Reply:
Not quite the same - half of them are set_params the other half use get_params.. So unless you want goto's everywhere that wont work :)